### ISSUE TITLE
Can it be licensed under the MIT?

### ISSUE THREAD

**Issue by franciscop** [liferay/alloy-editor#666]
Hi everyone, I am the creator of [legally](https://github.com/franciscop/legally) but still not a lawyer (and IANAL and all). However, I've been searching around around using the LGPL with MIT code and it seems it depends. From my interpretation and [reading this](http://softwareengineering.stackexchange.com/a/141848/71839) and [this](http://stackoverflow.com/questions/3902754/mit-vs-gpl-license), it seems that:

1. If I use a CDN or similar to **link** to the Alloy Editor, then it's fine.
2. However if I **include** the Alloy Editor within my code, I would have to license my code under the LGPL license.
3. If I **modify** the Alloy Editor code at all (even with minification), I would also have to license those changes with the LGPL.

Note that 2. and 3. would be totally normal in a concatenate and minimize toolchain that are almost a requirement in JS world for a good user experience. So, from my understanding of the licenses, there is no real way to include the Allow Editor in a performant way in MIT licensed code. Would you consider to switch to an MIT license?

**Comment by jbalsas** [comment]
Hey @franciscop, thanks for asking! This is a very interesting topic!

I'm not a lawyer either, but the main issue I see here is that AlloyEditor works on top of [CKEditor](http://ckeditor.com/) which is [licensed as LGPL](https://github.com/ckeditor/ckeditor-dev/blob/master/LICENSE.md) as well, so I don't think we could distribute it as MIT even if we wanted...

**Comment by ipeychev** [comment]
Hello,

Tl;DR I am also not a layer, but I don't see issues using LGPL code in conjunction with MIT code.

About 2., I don't think this applies to LGPL. It applies to GPL, so I don't see what prevents you from using LGPL source code in MIT. See [here](https://www.gnu.org/licenses/gpl-faq.en.html#LGPLv3ContributorVersion). You may distribute in two ways:
1. You include AlloyEditor's file using `script` element to your page, as it comes and use it.
2. If you are using Webpack or something else and install it from the NPM package, you may separate AlloyEditor from your code, which you have to do anyway if you care about performance. [Here](https://github.com/ipeychev/alloyeditor-react-component/blob/master/gulpfile.js#L18) is an example.
I also think if you include it in your code, but you keep the copyright message, that should be OK too.

About 3) - my personal opinion is that if you use Open Source software, the changes you make on this software should be also open sourced. The rest of your app program you may hide, but the modifications on LGPL software you have to open source. That's why I like LGPL.

**Comment by ipeychev** [comment]
@jbalsas CKEditor is licensed under multiple licenses - https://github.com/ckeditor/ckeditor-releases/blob/master/LICENSE.md

**Comment by jbalsas** [comment]
True, @ipeychev, I meant that none of them was MIT, though.

**Comment by ipeychev** [comment]
I see, sure/

**Comment by franciscop** [comment]
@ipeychev I agree with 3) and I always try to push the fixes upstream. I mean that technically, minimizing the code can be seen as a *modification* but it's quite pointless to require a 3rd party to publish the original code but just minimized.

If you depend on CKEditor and it's LGPL I see how the only option is for you to publish it as LGPL and not being able to change it. I close this issue then.
